### ISSUE TITLE
reflectable fails with prefixed annotations 

### ISSUE THREAD

**Issue by MikeMitterer** [dart-lang/reflectable#132]
```dart
import 'package:dryice/dryice.dart' as di;

@di.injectable
class MaterialApplication {
    void run() {}
}
```
What reflectable builder generates:
```dart
import "dart:core";
import 'dart:async' as prefix10;
import 'dart:html' as prefix9;
import 'package:dryice/dryice.dart' as prefix0;
...

       new r.NonGenericClassMirrorImpl(
            r"MaterialApplication",
            r"mdlcore.MaterialApplication",
            7,
            1,
            const prefix0.InjectAnnotation(),
            const <int>[20, 21],
            const <int>[15, 16, 17, 18, 19, 20],
            const <int>[],
            -1,
            {},
            {},
            {r"": (b) => () => b ? new prefix2.MaterialApplication() : null},
            1,
            1,
            const <int>[],
            const <Object>[prefix0.di.injectable],
            null),
```
This is where the problem occures: `const <Object>[prefix0.di.injectable],`
double prefix...

**Comment by MikeMitterer** [comment]
Sorry - forgot the error-message
Shows something like
```
[error] The values in a const list literal must be constants. (/test/unit/directive/components/Translate_test.reflectable.dart, line 24, col 4496)
[error] The values in a const list literal must be constants. (/test/unit/directive/components/Translate_test.reflectable.dart, line 24, col 4496)
[error] The getter 'di' isn't defined for the class 'prefix0'. (/test/unit/directive/components/Translate_test.reflectable.dart, line 24, col 4504)
```

**Comment by eernstg** [comment]
The following test (which uses a reflector via an import prefix) does not trigger this bug: https://github.com/dart-lang/test_reflectable/blob/master/test/prefixed_annotation_test.dart.

@MikeMitterer, do you know in which way we'd need to extend that example in order to make the bug happen?

**Comment by MikeMitterer** [comment]
@eernstg The branch `prefixedReflection` in https://github.com/MikeMitterer/objectvalidator.dart/tree/prefixedReflection shows the problem.

In particular this file:
https://github.com/MikeMitterer/objectvalidator.dart/blob/prefixedReflection/lib/objectvalidator.dart#L35

```bash
pub run build_runner build
[INFO] Generating build script completed, took 671ms
[WARNING] Throwing away cached asset graph because the build phases have changed. This most commonly would happen as a result of adding a new dependency or updating your dependencies.
[INFO] Cleaning up outputs from previous builds. completed, took 393ms
[INFO] Reading cached asset graph completed, took 920ms
[INFO] Building new asset graph completed, took 1.4s
[INFO] Checking for unexpected pre-existing outputs. completed, took 1ms
[SEVERE] build_web_compilers|ddc on test/unit/_resources/TestClasses.dartdevc.module:
Error compiling dartdevc module:objectvalidator|test/unit/_resources/TestClasses.ddc.js

[error] The values in a const list literal must be constants. (/test/unit/objectvalidator/ObjectValidator_test.reflectable.dart, line 2455, col 28)
[error] The values in a const list literal must be constants. (/test/unit/objectvalidator/ObjectValidator_test.reflectable.dart, line 2455, col 28)
[error] The getter 'di' isn't defined for the class 'prefix2'. (/test/unit/objectvalidator/ObjectValidator_test.reflectable.dart, line 2455, col 36)

Please fix all errors before compiling (warnings are okay).
}
```

The class `ReflectionSample` has no purpose it is just here for demonstrating the problem

**Comment by MikeMitterer** [comment]
Beside these little glitches your package works great!

**Comment by eernstg** [comment]
Thanks, and thanks! Looking into it.

**Comment by eernstg** [comment]
Try this:
```
  reflectable:
    git:
      url: git://github.com/dart-lang/reflectable.git
      ref: issue_132_sep18
```
It's under review, is expected to be published as 2.0.7 soon.
